src/lexer.coffee

```diff
@@ -1023,10 +1055,18 @@ exports.Lexer = class Lexer
     if lineCount > 0
       [..., lastLine] = string.split '\n'
       column = lastLine.length
+      previousLinesCompensation = @getLocationDataCompensation @chunkOffset, @chunkOffset + offset - column
+      # Don't recompensate for initially inserted newline.
+      previousLinesCompensation = 0 if previousLinesCompensation < 0
+      columnCompensation = @getLocationDataCompensation(
```

Comment:
This gets a little hairy algorithmically but seems to be the correct behavior - if we're calculating the compensation for a column number of a chunk that includes a newline, calculate the compensation for just the part after the last newline, adjusting for the compensations in the part of the chunk *before* the last newline